e2e-polybft/e2e/bridge_test.go

```diff
@@ -1442,3 +1442,90 @@ func TestE2E_Bridge_NonMintableERC20Token_WithPremine(t *testing.T) {
 		}
 	})
 }
+
+func TestE2E_Bridge_NonNative(t *testing.T) {
+	const (
+		transfersCount       = 5
+		amount               = 100
+		epochSize            = 5
+		numberOfAttempts     = 4
+		stateSyncedLogsCount = 2
+	)
+
+	var (
+		stakeAmount = ethgo.Ether(500)
+	)
+
+	minter, err := wallet.GenerateKey()
+	require.NoError(t, err)
+
+	cluster := framework.NewTestCluster(t, 5,
+		framework.WithNumBlockConfirmations(0),
+		framework.WithEpochSize(epochSize),
+		framework.WithBridge(),
+		framework.WithBladeAdmin(minter.Address().String()),
+		framework.WithSecretsCallback(func(addrs []types.Address, tcc *framework.TestClusterConfig) {
+			for i := 0; i < len(addrs); i++ {
+				tcc.StakeAmounts = append(tcc.StakeAmounts, stakeAmount)
+			}
+		}),
+		framework.WithPredeploy(),
+	)
+	defer cluster.Stop()
+
+	cluster.WaitForReady(t)
+
+	firstValidator := cluster.Servers[0]
+
+	validatorAcc, err := validatorHelper.GetAccountFromDir(firstValidator.DataDir())
+	require.NoError(t, err)
+
+	relayer, err := txrelayer.NewTxRelayer(txrelayer.WithIPAddress(firstValidator.JSONRPCAddr()))
+	require.NoError(t, err)
+
+	mintFn := &contractsapi.MintRootERC20Fn{
+		To:     validatorAcc.Address(),
+		Amount: big.NewInt(100000),
+	}
+
+	mintInput, err := mintFn.EncodeAbi()
+	require.NoError(t, err)
+
+	nonNativeErc20 := ethgo.Address(contracts.ERC20Contract)
```

Comment:
You shouldn't hardcode address to ERC20Contract, but use the one written in the `PolyBFTConfig`.